kinto/api-test/mockclient.py

```diff
@@ -40,6 +39,13 @@ def post_request(self, resource, data=None, status_code=None):
             r.raise_for_status()
         return json.loads(r.content.decode('utf-8'))
 
+    def post_json_request(self, resource, data=None, status_code=None):
```

Comment:
The json.dumps() method breaks other tests. The alternative is for me to
just manually use the requests library instead of adding a new method.
Either way I think we need at least one test that submits Unicode
strings to make sure Kinto handles them correctly



On Tue, Dec 1, 2015, at 02:57 PM, stuartphilp wrote:
> In kinto/api-test/mockclient.py[1]:


> > @@ -40,6 +39,13 @@ def post_request(self, resource, data=None,
> > status_code=None):
>              r.raise_for_status() return json.loads(r.content.decode('utf-
>                8'))
>
> +    def post_json_request(self, resource, data=None,
>      status_code=None): does this need to be a separate request type?
>      maybe we can use the json.dumps method as in
>      http://docs.python-requests.org/en/latest/user/quickstart/#more-complicated-post-requests


> — Reply to this email directly or view it on GitHub[2].


>



Links:

  1. https://github.com/mozilla-services/services-test/pull/31#discussion_r46330028
  2. https://github.com/mozilla-services/services-test/pull/31/files#r46330028